packages/react/src/hooks/useParticipantTile.ts

```diff
@@ -39,10 +39,16 @@ export function useParticipantTile<T extends HTMLElement>({
   const maybeTrackRef = useMaybeTrackRefContext();
   const p = useEnsureParticipant(participant);
   const trackReference = React.useMemo(() => {
+    const _source = trackRef?.source ?? maybeTrackRef?.source ?? source;
+    if (_source === undefined) {
+      throw new Error(
+        'Missing track `source`, provided it via `trackRef`, `source` property or via `TrackRefContext`.',
```

Comment:
just a thought: with all the deprecation code we're currently writing, it might already be beneficial to have a custom error for this in order to not have to write out this whole string all the time.

Reply:
Good idea. Expanding on that, we could have some predefined errors for missing contexts with helpful error messages and how to fix the problem. Maybe even a link to the docs.